lib/apollo-exception.filter.ts

```diff
@@ -0,0 +1,39 @@
+import { Catch, ArgumentsHost, HttpException } from '@nestjs/common';
+import {
+  ApolloError,
+  AuthenticationError,
+  ForbiddenError,
+} from 'apollo-server-errors';
+import { GqlExceptionFilter } from './interfaces';
+import { GqlContextType } from './services';
+
+const apolloPredefinedExceptions: Record<number, typeof ApolloError> = {
+  401: AuthenticationError,
+  403: ForbiddenError,
+};
+
+@Catch(HttpException)
+export class ApolloExceptionFilter implements GqlExceptionFilter {
+  catch(exception: HttpException, host: ArgumentsHost) {
+    if (host.getType<GqlContextType>() !== 'graphql') {
+      throw exception;
+    }
+
+    let error: ApolloError;
+    if (exception.getStatus() in apolloPredefinedExceptions) {
+      error = new apolloPredefinedExceptions[exception.getStatus()](
+        exception.message,
+      );
+    } else {
+      error = new ApolloError(
+        exception.message,
+        exception.getStatus().toString(),
+      );
+    }
+
+    error.stack = exception.stack;
+    error.extensions['response'] = exception.getResponse();
+
+    throw error;
```

Comment:
Based on the `GqlExceptionFilter ` interface you should return here.

Reply:
That's on purpose though. I want ApolloServer to catch this exception so it can format it nicely in the response.